test/test_t_test.cpp

```diff
@@ -95,6 +95,53 @@ void test_agreement_with_mathematica()
     }
 }
 
+template<typename Real>
+void test_two_sample_t()
+{
+    std::vector<Real> set_1 {301, 298, 295, 297, 304, 305, 309, 298, 291, 299, 293, 304};
+    std::vector<Real> set_2 {302, 309, 324, 313, 312, 310, 305, 298, 299, 300, 289, 294};
+
+    // https://www.statology.org/two-sample-t-test-calculator/
+    const double expected_statistic = -1.608761;
+    const double expected_pvalue = 0.121926;
+
+    auto [computed_statistic, computed_pvalue] = boost::math::statistics::two_sample_t_test(set_1, set_2);
+
+    CHECK_MOLLIFIED_CLOSE(expected_statistic, computed_statistic, 0.00001);
+    CHECK_MOLLIFIED_CLOSE(expected_pvalue, computed_pvalue, 0.00001);
+}
```

Comment:
This is a good test, but is there more symmetric data for which you can compute an exact solution by hand? I tend to find those more compelling.

Also, the precision of the calculation should increase with the precision of the `Real` type, but yours does not. I would use `CHECK_ULP_CLOSE`.